components/prize_pool/commons/prize_pool_import.lua

```diff
@@ -542,18 +543,38 @@ function Import._groupLastVsAdditionalData(lpdbEntry)
 		table.insert(matchConditions, '[[match2id::' .. matchId .. ']]')
 	end
 
-	local matchData = mw.ext.LiquipediaDB.lpdb('match2', {
-		conditions = '[[opponent::' .. opponentName .. ']] AND (' .. table.concat(matchConditions, ' OR ') .. ')',
-		order = 'date desc, match2id desc',
-		query = 'date, match2opponents, winner',
-		limit = 1
-	})
+	if table.concat(matchConditions, ' OR ') ~= _last_vs_group_cache.conditions then
+		_last_vs_group_cache.conditions = table.concat(matchConditions, ' OR ')
+		Import._lastVsMatchesDataToCache(mw.ext.LiquipediaDB.lpdb('match2', {
+			conditions = _last_vs_group_cache.conditions,
+			order = 'date desc, match2id desc',
+			query = 'date, match2opponents, winner',
+			limit = 1000,
+		}))
+	end
+
+	local matchData = _last_vs_group_cache.data[opponentName]
 
-	if not type(matchData) == 'table' or not matchData[1] then
+	if not matchData then
 		return {}
 	end
 
-	return Import._makeAdditionalDataFromMatch(opponentName, matchData[1])
+	return Import._makeAdditionalDataFromMatch(opponentName, matchData)
+end
+
+function Import._lastVsMatchesDataToCache(queryData)
+	local byOpponent = {}
+
+	for _, match in ipairs(queryData) do
+		for _, opponent in pairs(match.match2opponents) do
+			local name = opponent.name
+			if not byOpponent[name] then
+				byOpponent[name] = match
+			end
+		end
+	end
+
+	_last_vs_group_cache.data = byOpponent
```

Comment:
Why only cache one match?

Reply:
we only want the last played match for lastVs, hence only caching the last match for each opponent

due to the sort of the lpdb query the first match it finds is the last match they played

Reply:
It's not caching the last match for each opponent, it's just caching the last match seen

Reply:
it is caching per opponent (up to) 1 match

it runs over all matches and if for an opponent (name) there is no match yet it caches that match for said opponent

this way it should cache the last match played per opponent